chia/util/api_decorators.py

```diff
@@ -1,29 +1,21 @@
 from __future__ import annotations
 
-import functools
 import logging
 from dataclasses import dataclass, field
-from inspect import signature
-from typing import TYPE_CHECKING, Any, Callable, Coroutine, List, Optional, Union, get_type_hints
+from typing import Any, Callable, List, Optional, TypeVar, Union, get_type_hints
+
+from typing_extensions import Concatenate, ParamSpec
 
 from chia.protocols.protocol_message_types import ProtocolMessageTypes
-from chia.server.outbound_message import Message
-from chia.util.streamable import Streamable, _T_Streamable
+from chia.util.streamable import Streamable
 
 log = logging.getLogger(__name__)
 
-if TYPE_CHECKING:
-    from chia.server.ws_connection import WSChiaConnection
-
-    converted_api_f_type = Union[
-        Callable[[Union[bytes, _T_Streamable]], Coroutine[Any, Any, Optional[Message]]],
-        Callable[[Union[bytes, _T_Streamable], WSChiaConnection], Coroutine[Any, Any, Optional[Message]]],
-    ]
+P = ParamSpec("P")
+R = TypeVar("R")
+S = TypeVar("S", bound=Streamable)
```

Comment:
```suggestion
S = TypeVar("S", bound=Streamable)
```

Why are you dropping the one from the streamable module? We could just make it public there maybe?

Reply:
I don't have a fully formed plan or policy around handling these `TypeVar`s, but I don't think that at least trivial ones like this benefit much from being shared.  Happy to chat about it towards the end of coming up with preferred practices and their reasons.